### PR TITLE
Added endpoint for deactivating third party credentials

### PR DESCRIPTION
**Pull request by Spartakovic**
We need an endpoint that we can `POST` to which will deactivate a users third-party (Google/Github/Twitter) account from atlas admin.

### CODE REVIEW

File: atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java [hunk 1]
```diff
@@ -146,6 +155,33 @@ public void updateUser(HttpServletRequest request,
             new ErrorResultWriter().write(summary, writer, request, response);
         }
     }
+
+    @RequestMapping(value = "/4/users/deactivate/{uid}.*", method = RequestMethod.POST)
+    public void deactivateUser(HttpServletRequest request,
+            HttpServletResponse response,
+            @PathVariable String uid) throws IOException {
+        ResponseWriter writer = null;
```
Comment (JamieAP) [inline atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java]: This is unused, so why make it?

**Comment by JamieAP** [comment]
Otherwise LGTM

File: atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java (hunk 1 above)
Comment (Spartakovic) [inline atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java]: it is used for ErrorSummary)

File: atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java [hunk 2]
```diff
@@ -146,6 +155,32 @@ public void updateUser(HttpServletRequest request,
...
+        try {
+            User editingUser = userFetcher.userFor(request).get();
+            if (!editingUser.is(Role.ADMIN)) {
+                throw new ResourceForbiddenException();
+            }
+            Id userId = Id.valueOf(idCodec.decode(uid));
```
Comment (cognitive-i) [inline atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java]: Some whitespace would be helpful here.

File: atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java [hunk 3]
```diff
@@ -146,6 +155,32 @@ public void updateUser(HttpServletRequest request,
...
+    @RequestMapping(value = "/4/users/deactivate/{uid}.*", method = RequestMethod.POST)
+    public void deactivateUser(HttpServletRequest request,
+            HttpServletResponse response,
+            @PathVariable String uid) throws IOException {
+        ResponseWriter writer = null;
+        try {
+            User editingUser = userFetcher.userFor(request).get();
+            if (!editingUser.is(Role.ADMIN)) {
+                throw new ResourceForbiddenException();
+            }
```
Comment (cognitive-i) [inline atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java]: Some whitespace would be helpful here

File: atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java [hunk 4]
```diff
@@ -65,6 +73,7 @@ public UsersController(UserAwareQueryParser<User> requestParser,
         this.userFetcher = userFetcher;
         this.userStore = userStore;
         this.clock = clock;
+        this.credentialsStore = (MongoDBCredentialsStore) credentialsStore;
```
Comment (emilssolmanis) [inline atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java]: Why do we need the cast? Should we actually be fixing the interface?

File: atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java [hunk 5]
```diff
@@ -146,6 +155,32 @@ public void updateUser(HttpServletRequest request,
...
+                Credentials credentials = credentialsStore.find(userRef).requireValue();
+                credentialsStore.expired(credentials.authToken());
+            } else {
+                throw new NotFoundException(userId);
+            }
+        } catch (Exception e) {
+            log.error("Request exception " + request.getRequestURI(), e);
+            ErrorSummary summary = ErrorSummary.forException(e);
+            new ErrorResultWriter().write(summary, writer, request, response);
+            }
```
Comment (emilssolmanis) [inline atlas-api/src/main/java/org/atlasapi/application/users/UsersController.java]: Misaligned whitespace.

**Pull request closed without merging**